Sources/SemanticAnalyzer/SemanticAnalyzer+Expressions.swift

```diff
@@ -134,19 +129,37 @@ extension SemanticAnalyzer {
 
     switch externalCall.mode {
     case .normal:
-      // Ensure `call` is only used inside do-catch block
-      if passContext.doBlockNestingCount <= 0 {
+      // Ensure `call` is only used inside the do-body of do-catch statement
+      if passContext.doCatchStatementStack.last != nil {
+        // Update containsExternallCall value of doCatchStatement
+        var doCatchStatementStack = passContext.doCatchStatementStack
+        doCatchStatementStack[doCatchStatementStack.count - 1].containsExternalCall = true
```

Comment:
Use doCatchStatementStack.last or conditionally bind in the if since this is the same expression

Reply:
`if var` copies by value and `doCatchStatement.last` is get-only

Reply:
not possible

Reply:
rip